starknet_py/tests/e2e/docs/guide/test_resolving_proxies.py

```diff
@@ -12,21 +28,54 @@ async def test_resolving_proxies(
     # docs: end
     address = map_contract.address
     # docs: start
-    # If the contract is not a proxy just pass its address and a client
+    # Getting the direct contract from address
     contract = await Contract.from_address(address=address, client=gateway_client)
 
     # docs: end
     address = deploy_proxy_to_contract_oz_argent.deployed_contract.address
     # docs: start
-    # However, if you have a proxy and want to use it as a regular contract,
-    # you will have to pass one more argument
+    # To use contract behind a proxy as a regular contract, set proxy_config to True
     # It will check if your proxy is OpenZeppelin or ArgentX proxy
-    # To resolve other proxies pass custom ProxyCheck as proxy_config parameter
     contract = await Contract.from_address(
         address=address, client=gateway_client, proxy_config=True
     )
 
     # After that contract can be used as usual
+
+    # To resolve proxy contract other than OpenZeppelin or ArgentX, a custom ProxyCheck is needed
+    # The ProxyCheck below resolves proxy contracts which have implementation
+    # stored in implementation() function as address
+    class CustomProxyCheck(ProxyCheck):
+        async def implementation_address(
+            self, address: Address, client: Client
+        ) -> Optional[int]:
+            return None
```

Comment:
A comment about why it is returning none could be useful here considering that it is in documentation.